### ISSUE TITLE
Should agents contain environments or vice versa?

### ISSUE THREAD

**Issue by nickjalbert** [agentos-project/agentos#73]
In our current model, an [agent contains an environment](https://agentos.org/latest/architecture_and_design.html#id2):

![agent_env2](https://user-images.githubusercontent.com/25208102/106623715-bfbf9380-6529-11eb-9fac-8a9780c31150.png)

This makes sense for implementation reasons, but (I think) this is the reverse of the intuitive way one would talk about these things (i.e. an agent performs behavior within an environment and an environment contains one or more agents).

This issue is a placeholder to remind us to discuss the tradeoff of modelling it the current way as opposed to the inverse way.

**Comment by nickjalbert** [comment]
Not sure if filing issues is a good way to track these "higher-level" discussions; @andyk let me know if you'd prefer some other approach (I could just keep a list for planning meeting or something).

**Comment by andyk** [comment]
Yeah an issue seems like a great way to track big picture things like this.

I agree that it is a counter intuitive to have an agent contain an env and not vice verse.

It would be great to come up with a few concrete alternatives and then weigh the trade-offs between them.
